exporters/otlp/include/opentelemetry/exporters/otlp/otlp_http_client.h

```diff
@@ -98,14 +104,16 @@ class OtlpHttpClient
    * Create an OtlpHttpClient using the given options.
    */
   explicit OtlpHttpClient(OtlpHttpClientOptions &&options);
+#ifdef ENABLE_TEST
+  VIRTUAL_TEST ~OtlpHttpClient() {}
+#endif
 
   /**
    * Export
    * @param message message to export, it should be ExportTraceServiceRequest,
    * ExportMetricsServiceRequest or ExportLogsServiceRequest
    */
-  sdk::common::ExportResult Export(const google::protobuf::Message &message) noexcept;
-
+  VIRTUAL_TEST sdk::common::ExportResult Export(const google::protobuf::Message &message) noexcept;
```

Comment:
`googletest` suggests to mock the non-virtual function using template - https://chromium.googlesource.com/external/github.com/google/googletest/+/refs/tags/release-1.8.0/googlemock/docs/CookBook.md#mocking-nonvirtual-methods, we can avoid using ENABLE_TEST macro in that case.